WorkloadManagementSystem/Agent/SiteDirector.py

```diff
@@ -610,456 +71,11 @@ def __getQueueSlots( self, queue ):
-    if not result[ 'OK' ]:
-      self.log.error( 'Invalid proxy token request', result['Message'] )
-      return [ None, None ]
-    ( token, numberOfUses ) = result[ 'Value' ]
-    pilotOptions.append( '-o /Security/ProxyToken=%s' % token )
-    # Use Filling mode
-    pilotOptions.append( '-M %s' % min( numberOfUses, self.maxJobsInFillMode ) )
-
-    # Since each pilot will execute min( numberOfUses, self.maxJobsInFillMode )
-    # with numberOfUses tokens we can submit at most:
-    #    numberOfUses / min( numberOfUses, self.maxJobsInFillMode )
-    # pilots
-    newPilotsToSubmit = numberOfUses / min( numberOfUses, self.maxJobsInFillMode )
-    if newPilotsToSubmit != pilotsToSubmit:
-      self.log.info( 'Number of pilots to submit is changed to %d after getting the proxy token' % newPilotsToSubmit )
-      pilotsToSubmit = newPilotsToSubmit
-    # Debug
-    if self.pilotLogLevel.lower() == 'debug':
-      pilotOptions.append( '-d' )
-    # CS Servers
-    csServers = gConfig.getValue( "/DIRAC/Configuration/Servers", [] )
-    pilotOptions.append( '-C %s' % ",".join( csServers ) )
-
-    # DIRAC Extensions to be used in pilots
-    pilotExtensionsList = opsHelper.getValue( "Pilot/Extensions", [] )
-    extensionsList = []
-    if pilotExtensionsList:
-      if pilotExtensionsList[0] != 'None':
-        extensionsList = pilotExtensionsList
-    else:
-      extensionsList = CSGlobals.getCSExtensions()
-    if extensionsList:
-      pilotOptions.append( '-e %s' % ",".join( extensionsList ) )
-
-    # Requested CPU time
-    pilotOptions.append( '-T %s' % queueDict['CPUTime'] )
-    # CEName
-    pilotOptions.append( '-N %s' % self.queueDict[queue]['CEName'] )
-    # Queue
-    pilotOptions.append( '-Q %s' % self.queueDict[queue]['QueueName'] )
-    # SiteName
-    pilotOptions.append( '-n %s' % queueDict['Site'] )
-    if 'ClientPlatform' in queueDict:
-      pilotOptions.append( "-p '%s'" % queueDict['ClientPlatform'] )
-
-    if 'SharedArea' in queueDict:
-      pilotOptions.append( "-o '/LocalSite/SharedArea=%s'" % queueDict['SharedArea'] )
-
-#     if 'SI00' in queueDict:
-#       factor = float( queueDict['SI00'] ) / 250.
-#       pilotOptions.append( "-o '/LocalSite/CPUScalingFactor=%s'" % factor )
-#       pilotOptions.append( "-o '/LocalSite/CPUNormalizationFactor=%s'" % factor )
-#     else:
-#       if 'CPUScalingFactor' in queueDict:
-#         pilotOptions.append( "-o '/LocalSite/CPUScalingFactor=%s'" % queueDict['CPUScalingFactor'] )
-#       if 'CPUNormalizationFactor' in queueDict:
-#         pilotOptions.append( "-o '/LocalSite/CPUNormalizationFactor=%s'" % queueDict['CPUNormalizationFactor'] )
-
-    if "ExtraPilotOptions" in queueDict:
-      pilotOptions.append( queueDict['ExtraPilotOptions'] )
-
+    pilotOptions, pilotsToSubmit = OriginalSiteDirector._getPilotOptions(self, queue, pilotsToSubmit)
     # Get the module specific options
     if self.extraOptions:
       pilotOptions.append( self.extraOptions )
```

Comment:
This of course relies on the assumption that the options order doesn't matter to greatly reduce our SiteDirector code. Seems this assumption is ok in @sfayer 's tests